server/controllers/users.js

```diff
@@ -282,3 +284,49 @@ export function getCallCompleteTwiml(req, res) {
     .set({ 'Content-Type': 'text/xml' })
     .send(callCompletedTwiml);
 }
+
+function getContactIdIfNotExist(callId, contactId) {
+  return new Promise((resolve, reject) => {
+    if (!contactId) {
+      callsService.getCallById({ id: callId })
+      .then((call) => {
+        const { contact_id } = call;
+        resolve({ contact_id });
+      })
+      .catch(err => reject(err));
+    } else {
+      resolve({ contact_id: contactId });
+    }
+  });
+}
+
+export function connectVolunteerToContact(req, res) {
+  const user_id = parseInt(req.params.id, 10);
+  const campaign_id = parseInt(req.params.campaign_id, 10);
+  const call_id = parseInt(req.params.call_id, 10);
+  const { contact_id: contactId } = req.body;
```

Comment:
Does the API spec for the ticket suggest that the `contact_id` param will be present? If so, why conditionally fetch it with getContactIdIfNotExist - we should fail the request if it's missing? And if it's not in the API, why bother checking, just always get it via the calls service? This will also let us get rid of the nasty typeof check we do in line 312.

Reply:
I figured it could be sent as a body param, but since I've gotten a better idea of the big picture, I'll just fetch the contact id with the service.